metadata_mapper/mappers/mapper.py

```diff
@@ -231,16 +237,16 @@ def shred(self, field, delim=";"):
         Duplicate values are removed.
 
         called with the following parameters:
-        2,078 times:    field=["sourceResource/spatial"],       delim = ["--"]
-        1,021 times:    field=["sourceResource/subject/name"]
-        1,021 times:    field=["sourceResource/creator"]
-        1,021 times:    field=["sourceResource/type"]
-        5 times:        field=["sourceResource/description"],   delim=["<br>"]
+        2,078 times:    prop=["sourceResource/spatial"],       delim = ["--"]
+        1,021 times:    prop=["sourceResource/subject/name"]
+        1,021 times:    prop=["sourceResource/creator"]
+        1,021 times:    prop=["sourceResource/type"]
+        5 times:        prop=["sourceResource/description"],   delim=["<br>"]
         """
-        field = field[0].split('/')[1:]     # remove sourceResource
+        field = prop[0].split('/')[1:][0]     # remove sourceResource
         delim = delim[0]
 
-        if field not in self.mapped_data:
+        if field not in self.mapped_data or not self.mapped_data[field]:
```

Comment:
sometimes field exists in self.mapped_data but self.mapped_data[field] literally equals None. Would love to understand better if there are cases where a "None" field value is actually important to know, or if this is just a side effect of implementation and we can safely clean up these fields.

Reply:
Apparently Lucene (which is what solr and elasticsearch are both built on) cannot by definition index a null value. It could index the string "None", but I'm doubtful that this is ever what is intended? We could check with Adrian?